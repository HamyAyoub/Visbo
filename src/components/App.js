import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import Landing from './landing/landing';
import Demo from './demo/demo';
import DemoBoard from './demo/demo-board';
import { connect } from 'react-redux';

function mapStateToProps(state) {
  return {
    todos: state.todos,
  };
}

class App extends Component {
  render() {
    return (
      <Switch>
        <Route exact path='/' >
          <Landing />
        </Route>

        <Route path='/demo' >
          <Demo />
        </Route>

        <Route path='/demo-board' >
          <DemoBoard />
        </Route>
      </Switch>
    );
  }
}
export default connect(mapStateToProps)(App);
