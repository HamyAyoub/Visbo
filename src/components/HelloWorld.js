import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

function mapStateToProps(state) {
  return {
    text: state.text
  };
}

class HelloWorld extends Component {
  render() { 
    let { text } = this.props
    return ( 
      <h1>{text}</h1>
     );
  }
}

HelloWorld.propTypes = {
  text: PropTypes.string
}
 
export default connect(mapStateToProps)(HelloWorld);