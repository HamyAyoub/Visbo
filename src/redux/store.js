import { createStore } from "redux";

const initialState = {
  projects: [
    {
      id: 0,
      title: 'Wash dishes',
    },
    {
      id: 1,
      title: 'Do homework',
    },
    {
      id: 3,
      title: 'Buy groceries',
    }
  ]
}

function reducer(state = initialState, action) {
  return state;
}

const store = createStore(reducer);

export default store;
