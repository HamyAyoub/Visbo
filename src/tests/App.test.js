import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import HelloWorld from '../components/HelloWorld';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../redux/store';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

configure({ adapter: new Adapter() })

// function shallowSetup() {
//   const props = {
//    text: "Hello World"
//   }

//   const enzymeWrapper = shallow(
//     <Provider store={store}>
//         <HelloWorld />
//     </Provider>
//   ).dive();

//   return {
//     props,
//     enzymeWrapper
//   };
// }

describe("HelloWorld Component", () => {

  it('renders without crashing using jest', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <HelloWorld />
      </Provider>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <HelloWorld />
      </Provider>
      ).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('should render an h1 tag containing Hello World string', () => {
    //const { enzymeWrapper, props } = shallowSetup();

    const enzymeWrapper = shallow(
      <Provider store={store}>
        <HelloWorld />
      </Provider>
    )

    expect(enzymeWrapper.find('h1').text()).toBe(props.text);
   // expect(enzymeWrapper.containsMatchingElement(<h1>Hello World</h1>)).toBe(true);
  });
})



