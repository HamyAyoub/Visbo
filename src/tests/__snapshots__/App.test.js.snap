// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld Component renders the UI as expected 1`] = `
<h1>
  Hello World
</h1>
`;
